test/Dialect/FIRRTL/canonicalization.mlir

```diff
@@ -2293,6 +2293,21 @@ firrtl.module @ForceableRegResetToNode(in %clock: !firrtl.clock, in %dummy : !fi
   firrtl.connect %foo, %reg: !firrtl.uint<1>, !firrtl.uint<1>
 }
 
+  // https://github.com/llvm/circt/issues/8348
+  // CHECK-LABEL: firrtl.module @RegResetInvalidResetValueType
+  // We cannot replace a regreset with its reset value, when the reset value's type does not match.
+  firrtl.module @RegResetInvalidResetValueType(out %out : !firrtl.uint<2>) {
+    %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+    %c0_ui2 = firrtl.constant 0 : !firrtl.uint<2>
+    %c0_clock = firrtl.specialconstant 0 : !firrtl.clock
```

Comment:
Maaaaybe feed clock from a block argument?

I don't know but "a register that we know is never clocked" -> (??) might trigger a different canonicalizer that's not the one intended to be tested here.

Reply:
Done, thanks :+1: